src/blob/index/bptree/core.rs

```diff
@@ -131,6 +131,12 @@ where
                 }
                 Ok(headers)
             })
+            .map(|mut headers| {
+                for val in headers.values_mut() {
+                    val.reverse();
```

Comment:
For the most common case when `val.len() == 1` we can avoid calling `reverse`. Please, add condition before the call

Reply:
I'm surprised this check is not part of the reverse itself. Ok, I'll add this

Reply:
According to the source code (https://doc.rust-lang.org/src/core/slice/mod.rs.html#678) it is not a part